logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/message/ReceiptMessageHandler.kt

```diff
@@ -0,0 +1,31 @@
+package com.wire.kalium.logic.sync.receiver.message
+
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.receipt.ReceiptRepository
+import com.wire.kalium.logic.data.user.UserId
+import kotlinx.datetime.Instant
+
+internal class ReceiptMessageHandler(
+    private val selfUserId: UserId,
+    private val receiptRepository: ReceiptRepository
+) {
+
+    suspend fun handle(
+        message: Message.Signaling,
+        messageContent: MessageContent.Receipt
+    ) {
+        // Receipts from self user shouldn't happen,
+        // If it happens, it's unnecessary,
+        // and we can squish some performance by skipping it completely
+        if (message.senderUserId == selfUserId) return
```

Comment:
Although there are more rules we need to follow, like making sure the IDs exist, etc. the Database constraints take care of this.